### PR TITLE
feat: Action Widgets Reskinning - Button, Button Group, Icon Button, Menu Button

### LINKED ISSUE #10972: [Enhancement] : Button widget reskinned
https://www.notion.so/appsmith/Button-bb3d820d41da428db9a98a4fa1f88afc

### LINKED ISSUE #10989: [Enhancement]: Icon button widget reskinning
https://www.notion.so/appsmith/Icon-button-9e58a97721c84a8f8ca03008b24353c3

### LINKED ISSUE #10996: [Enhancement]: Menu button widget reskinning
https://www.notion.so/appsmith/Menu-button-89d769789aac4cb9a374ea2c1ddbe21e

### LINKED ISSUE #15170: Button group reskinning
https://www.notion.so/appsmith/Button-group-96bf174a442d45f1bd15375e780afbc0

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description
This PR fixes UI inconsistencies in the following Action Widgets -
1. Button
2. Button Group
3. Icon Button
4. Menu Button

Fixes #10972
Fixes #15170 
Fixes #10989 
Fixes #10996

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/ButtonWidget/index.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const CONFIG = {
     text: "Submit",
     buttonVariant: ButtonVariantTypes.PRIMARY,
     placement: ButtonPlacementTypes.CENTER,
-    rows: 4,
+    rows: 5,
```
Comment (jsartisan) [inline app/client/src/widgets/ButtonWidget/index.ts:20]: @dilippitchika @dhruvikn Are we sure we want to increase button height? It looks to me as compared to standard button sizes.

File: app/client/src/widgets/IconButtonWidget/component/index.tsx [hunk 2]
```diff
@@ -155,17 +156,30 @@ export const StyledButton = styled((props) => (
...
+      background: ${
+        buttonVariant !== ButtonVariantTypes.TERTIARY
+          ? "var(--wds-color-bg-disabled)"
+          : "transparent"
+      } !important;
+      color: var(--wds-color-text-disabled) !important;
       pointer-events: none;
     }
 
     &&:disabled {
```
Comment (jsartisan) [inline app/client/src/widgets/IconButtonWidget/component/index.tsx:168]: Not a review but a question: do you know why we are using `&&:disabled`? we could have written the styles in just `&:disabled`

Comment (dhruvikn) [reply app/client/src/widgets/IconButtonWidget/component/index.tsx:168]: Hey, not completely sure, but it looks like here it is used to increase specificity (give precedence boost). Here are a few helpful links I used to understand what `&&` does. Will try removing this once and see what happens. If it's not affecting the disabled states, then let's merge both of them.

1. https://spectrum.chat/styled-components/general/how-does-the-ampersand-that-increases-specificity-work~89c28653-e27c-49f0-9d8c-d89dc05c59dc
2. https://styled-components.com/docs/basics#pseudoelements-pseudoselectors-and-nesting

File: app/client/src/widgets/ButtonWidget/index.ts (hunk 1 above)
Comment (dhruvikn) [reply app/client/src/widgets/ButtonWidget/index.ts:20]: It is also worth noting that the current height of buttons with row = 6 is 32px, and not 30px, as mentioned in the PRD.
Therefore, when changing row to 5, we will get a button with a height of 42px, not 40px.

**Comment by laveena-en** [comment]
Tested and verified.

**Comment by dhruvikn** [comment]
Thanks a lot for the quick QA @laveena-en and @chandannkumar. Kudos to you folks. 🙌 🚀

File: app/client/src/widgets/ButtonWidget/index.ts (hunk 1 above)
Comment (dhruvikn) [reply app/client/src/widgets/ButtonWidget/index.ts:20]: Had a discussion with @dilippitchika, @somangshu, and @jsartisan. We're gonna be sticking with row = 4 and button height = 32px. Will be reverting the change.

cc: @laveena-en @chandannkumar

**Review by jsartisan — DISMISSED** [review]
LGTM

**Review by jsartisan — APPROVED** [review]

**Pull request merged**